### PR TITLE
Fix bug with permissions

### LINKED ISSUE #121: Process list shows processes within Category that shouldn't be visible by user.
**Describe the bug**
For user, on the list of processes are visible all processes - even if he/she hasn't got configured Read permission for some category.

**To Reproduce**
1. Having Flink and Nussknacker ran e.g. in docker with configured two categories: one visible for user A and the second not visible.
2. Open browser, login as a user A and go to 'main page'

**Expected behavior**
User A should see processes in categories that he/she has Read permission,
, but:
User A see all processes.

### PR DESCRIPTION
**Pull request by lciolecki**
This PR closes #121 
 
- Remove Permission.Admin
- Add isAdmin property at user config file
- Add security for changing category, now this can do only admin
- Add some todos
- Some tests refactor

### CODE REVIEW

File: engine/security-api/src/test/scala/pl/touk/nussknacker/ui/security/api/PermissionSyntaxTest.scala [hunk 1]
```diff
@@ -21,14 +20,26 @@ class PermissionSyntaxTest extends FunSuite with Matchers {
       u.can(p) shouldEqual c
     }
   }
+
   test("Admin permission grands other permissions") {
-    PermissionSyntax.containsPermissionOrAdmin(Permission.Read)(Permission.Admin) shouldBe true
+    def admin(cp: Map[String, Set[Permission]]) = LoggedUser("admin", cp, Option.apply(true))
```
Comment (mproch) [inline engine/security-api/src/test/scala/pl/touk/nussknacker/ui/security/api/PermissionSyntaxTest.scala]: Option(true), and later Option(false)?

File: engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 
-import Permission.Permission
+import pl.touk.nussknacker.ui.security.api.Permission.Permission
 
-case class LoggedUser(id: String,
-                      categoryPermissions: Map[String, Set[Permission]]=Map.empty) {
+case class LoggedUser(
+  id: String,
+  categoryPermissions: Map[String, Set[Permission]] = Map.empty,
+  isAdmin: Option[Boolean] = Option.apply(false)
```
Comment (mproch) [inline engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala]: if we have default value does it make sense to have option here?

File: engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
+
   def hasPermission(permission: Permission): Boolean = {
-    permissions.contains(permission) || isAdmin
+    hasAdminPermission || permissions.contains(permission)
   }
 
-  //TODO: remove
-  def isAdmin: Boolean = permissions.contains(Permission.Admin)
-
+  def hasAdminPermission: Boolean = isAdmin.getOrElse(false)
```
Comment (mproch) [inline engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala]: maybe just isAdmin?

File: engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/PermissionSyntax.scala [hunk 4]
```diff
@@ -5,17 +5,16 @@ object PermissionSyntax {
   implicit class UserPermissionsQuery(user: LoggedUser) {
     def can(permission: Permission.Permission): Set[String] =
```
Comment (mproch) [inline engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/PermissionSyntax.scala]: If we make some cleaning maybe let's remove/change this method? It's a bit misleading in admin context - in fact for admin we want *all* categories and not only those to which admin has explicit permissions? See usages

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessRepository.scala [hunk 5]
```diff
@@ -17,8 +17,12 @@ trait ProcessRepository[F[_]] extends Repository[F] with EspTables {
   import api._
   import pl.touk.nussknacker.ui.security.api.PermissionSyntax._
   protected def processTableFilteredByUser(implicit loggedUser: LoggedUser): Query[ProcessEntityFactory#ProcessEntity, ProcessEntityData, Seq] = {
+    if (loggedUser.hasAdminPermission) {
```
Comment (mproch) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessRepository.scala]: let's not use explicit return and many exit points...

File: engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala (hunk 2 above)
Comment (lciolecki) [reply engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala]: Good idea :)

File: engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala (hunk 3 above)
Comment (lciolecki) [reply engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala]: Changed.

File: engine/security-api/src/test/scala/pl/touk/nussknacker/ui/security/api/PermissionSyntaxTest.scala (hunk 1 above)
Comment (lciolecki) [reply engine/security-api/src/test/scala/pl/touk/nussknacker/ui/security/api/PermissionSyntaxTest.scala]: Now is just LoggedUser("admin", cp, true).

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessRepository.scala (hunk 5 above)
Comment (lciolecki) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessRepository.scala]: Changed.

File: engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/PermissionSyntax.scala (hunk 4 above)
Comment (lciolecki) [reply engine/security-api/src/main/scala/pl/touk/nussknacker/ui/security/api/PermissionSyntax.scala]: I agree.. But if we want do this good, than we should create separated config which has all available categories.. It shoud be more complex solutions - now I want only fix it ;)

File: ui/server/develConf/sample/users.conf [hunk 6]
```diff
@@ -2,13 +2,8 @@ users: [
...
     id: "admin"
     password: "admin"
-    categoryPermissions: {
-      "Category1": ["Read", "Write", "Deploy", "Admin"]
-      "Category2": ["Read", "Write", "Deploy", "Admin"]
-      "Technical": ["Read", "Write", "Deploy", "Admin"]
-      "Default": ["Read", "Write", "Deploy", "Admin"]
-      "StandaloneCategory1": ["Read", "Write", "Deploy", "Admin"]
-    }
+    isAdmin: true
```
Comment (mproch) [inline ui/server/develConf/sample/users.conf]: I think we should fix it also in different places - e.g. docker-users.conf, etc?

Comment (lciolecki) [reply ui/server/develConf/sample/users.conf]: Ofc you are right :)

**Pull request merged**